korman/properties/modifiers/water.py

```diff
@@ -471,3 +471,36 @@ class PlasmaWaveTexState(PlasmaWaveState, PlasmaModifierProperties):
     def export(self, exporter, bo, so):
         waveset = exporter.mgr.find_create_object(plWaveSet7, name=bo.name, so=so)
         self.convert_wavestate(waveset.state.texState)
+
+
+class PlasmaBuoyObject(idprops.IDPropObjectMixin, bpy.types.PropertyGroup):
+    display_name = StringProperty(name="Display Name")
+    buoy_object = PointerProperty(name="Buoy Object",
+                                  description="Object that float on water",
+                                  type=bpy.types.Object,
+                                  poll=idprops.poll_mesh_objects)
+
+    @classmethod
+    def _idprop_mapping(cls):
+        return {"buoy_object": "object_name"}
```

Comment:
```suggestion
```

This is only needed for old string properties to be updated to pointer properties. No new property groups need it.